### ISSUE TITLE
The format of the RVC models has a .pt instead of .pth

### ISSUE THREAD

**Issue by jooni22** [sioaeko/OpenVoiceChanger#3]
Hello, there is little information about how to prepare a model to support it by the application, why python_server requires a model with an extension .pt instead .pth? The main RVC repository also shows how to train the RVC model where we finally get it as .pth, can you indicate more information on how to convert the model to .PT and .onnx?

**Comment by sioaeko** [comment]
To answer your question, the distinction between `.pt` and `.pth` files is typically related to PyTorch's model saving conventions. However, in essence, `.pt` and `.pth` are both used to store PyTorch models, and the difference is often more about naming convention than anything functional.

- `.pth` is commonly used to indicate the saved model's state dictionary (`state_dict`) of the model, while `.pt` can be used to store the complete model (including both architecture and weights) or just the model's state dictionary.
- The `.pt` extension is also sometimes used when you want to save a model for inference purposes, while `.pth` is often used during training.

### Converting `.pth` to `.pt`

If you have a model saved as `.pth`, you can easily convert it to `.pt` by saving the model in a `.pt` file, like so:

```python
import torch

# Assuming you have a trained model loaded as `model`
model = torch.load('model.pth')
torch.save(model, 'model.pt')  # Save the complete model in .pt format
```

Alternatively, if you're only saving the model's weights (the `state_dict`), you'd use the following:

```python
torch.save(model.state_dict(), 'model.pt')
```

### Converting to `.onnx`

To convert your PyTorch model (whether `.pt` or `.pth`) to ONNX, follow these steps:

1. **Prepare the model for export**: Ensure that the model is in evaluation mode (`model.eval()`), as ONNX export is designed to work with the model in inference mode.

2. **Export the model**:

```python
import torch.onnx

# Assuming `model` is your trained model
model.eval()  # Set the model to evaluation mode

# Create some dummy input matching the model's expected input shape
dummy_input = torch.randn(1, 3, 224, 224)  # Example for an image classification model

# Export the model to ONNX format
torch.onnx.export(model, dummy_input, "model.onnx")
```

This will create an ONNX file called `model.onnx` which can be loaded by other frameworks that support ONNX.

### Additional Information

- **Model Saving in RVC (Reconfigurable Vision and Control)**: The RVC repository might have specific scripts or methods for converting the model into a usable form for their application. Often, models trained for specific use cases like RVC might need some specific conversions or pre-processing, so make sure to check the repository’s documentation and any additional utilities or functions they provide for model conversion.
- **ONNX Support**: If the final goal is to use the model for deployment in an environment that requires ONNX, such as some edge devices or when using specific deployment tools, ONNX offers better compatibility across different hardware and frameworks (like TensorFlow or ONNX Runtime).

Let me know if you need any more details or help with a specific step!

**Comment by jooni22** [comment]
A really solid explanation, thank you very much!
